File: lessons/242/go-app/device.go
package main

type Device struct {
	Id        int    `json:"id"`
	Uuid      string `json:"uuid"`
	Mac       string `json:"mac"`
	Firmware  string `json:"firmware"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

File: lessons/233-elixir-vs-go-golang-performance-benchmark-round-2/elixir-app/lib/app/device.ex
defmodule App.Device do
  @derive {JSON.Encoder, only: [:id, :uuid, :mac, :firmware, :created_at, :updated_at]}
  defstruct [:id, :uuid, :mac, :firmware, :created_at, :updated_at]

  def save(device) do
    %{uuid: uuid, mac: mac, firmware: firmware, created_at: created_at, updated_at: updated_at} =
      device

    %{rows: [[id]]} =
      App.Repo.query!(
        "INSERT INTO \"devices\" (uuid, mac, firmware, created_at, updated_at) VALUES ($1, $2, $3, $4, $5) RETURNING id",
        [uuid, mac, firmware, created_at, updated_at]
      )

    %{device | id: id}
  end
end
